connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/FieldPath.java

```diff
@@ -0,0 +1,377 @@
+                            path); // avoid creating new string on changes
+
+                        while (s.length() > 0) { // until path is traverse
+                            // process backtick pair if any
+                            if (s.charAt(0) == BACKTICK_CHAR) {
+                                s.deleteCharAt(0);
+
+                                // find backtick pair
+                                int idx = 0;
+                                while (idx >= 0) {
+                                    idx = s.indexOf(BACKTICK, idx);
+                                    if (idx == -1) {
+                                        throw new IllegalArgumentException(
+                                            "Incomplete backtick pair at [...]`" + s);
+                                    }
+                                    if (idx != s.length() - 1) { // non-global backtick
+                                        if (s.charAt(idx + 1) != DOT_CHAR
+                                            || s.charAt(idx - 1)
+                                            == BACKSLASH_CHAR) { // not wrapped or escaped
+                                            idx++; // move index forward and keep searching
+                                        } else { // it's end pair
+                                            steps.add(
+                                                checkIncompleteBacktickPair(s.substring(0, idx)));
+                                            s.delete(0, idx + 2); // rm backtick and dot
+                                            break;
+                                        }
+                                    } else { // global backtick
+                                        steps.add(checkIncompleteBacktickPair(s.substring(0, idx)));
+                                        s.delete(0, s.length());
+                                        break;
+                                    }
+                                }
+                            } else { // process path dots
+                                final int atDot = s.indexOf(DOT);
+                                if (atDot > 0) { // get step and move forward
+                                    steps.add(checkIncompleteBacktickPair(s.substring(0, atDot)));
+                                    s.delete(0, atDot + 1);
+                                } else { // add all
+                                    steps.add(checkIncompleteBacktickPair(s.toString()));
+                                    s.delete(0, s.length());
+                                }
+                            }
+                        }
+
+                        this.path = steps.toArray(new String[0]);
+                    }
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown syntax version: " + version);
+            }
+        }
+    }
+
+    private String checkIncompleteBacktickPair(String field) {
```

Comment:
Would it be possible to add a doc comment for this method? Also `checkIncompleteBacktickPair` seems like a misnomer for a method that is returning a String

Reply:
Good call. I have renamed it to `escapeBackticks` and add docs.